Project.toml

```diff
@@ -1,44 +1,52 @@
 name = "ManifoldsBase"
 uuid = "3362f125-f0bb-47a3-aa74-596ffd7ef2fb"
 authors = ["Seth Axen <seth.axen@gmail.com>", "Mateusz Baran <mateuszbaran89@gmail.com>", "Ronny Bergmann <manopt@ronnybergmann.net>", "Antoine Levitt <antoine.levitt@gmail.com>"]
-version = "0.15.8"
+version = "0.15.9"
 
 [deps]
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 Markdown = "d6f4376e-aef5-505a-96c1-9c027394607a"
+Printf = "de0858da-6303-5e67-8744-51eddeeeb8d7"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
+Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
```

Comment:
These two new dependencies significantly increase loading time of Manifolds.jl for me. Current master branch loads for me in about 0.04s, while this branch takes a bit over 0.13s. I would say this is too much and would prefer to make these weak dependencies.

Reply:
Oh, I did not check for that. It might even be statistics, but I can not get rid of that, since that is used to compute the regression line.

Hm, making them weak is a bit of technical fuzz, including error hints and such, to me that feels like quite a bit of work, where Manifolds and Manopt have that as a dependency anyways.

But you also mean loading ManifoldsBase?

Reply:
Yes, that's mostly Statistics. I've checked it by executing `@time using ManifoldsBase`. Still, if some new package would like to depend on our interface, it might not need Statistics. If weak deps are too problematic then maybe let's put these checks in Manifolds.jl?

Reply:
Hm, I guess Statistics weakdep could be even useful here for reducing the amount of piracy Aqua detected in Manifolds.jl. So that probably would be the best way to go. Let me know if you need some help with that.

Reply:
Moving Statistics had the following effect on my machine:
Before: 0.0161 s
After:    0.0214s

I hope this is fine? Yes we then have Printf in the dependencies but I think that is not too bad.

Reply:
Great, thanks for changing Statistics to a weak dependency. 0.02s is still perfectly fine.